### ISSUE TITLE
FLAC: Decoder input buffer too small

### ISSUE THREAD

**Issue by tangmingscau** [google/ExoPlayer#3514]
I tried to be compatible with FLAC music with the ID3v2 head, and when I played the type of FLAC music, there were some mistakes. The following is the error log:

```
11-29 15:23:05.082 12598-12732/com.xiaopeng.musicradio E/ExoPlayerImplInternal: Internal runtime error.
  java.lang.IllegalStateException: Buffer too small (32768 < 49152)
    at com.google.android.exoplayer2.decoder.DecoderInputBuffer.createReplacementByteBuffer(DecoderInputBuffer.java:160)
    at com.google.android.exoplayer2.decoder.DecoderInputBuffer.ensureSpaceForWrite(DecoderInputBuffer.java:110)
    at com.google.android.exoplayer2.source.SampleQueue.read(SampleQueue.java:312)
    at com.google.android.exoplayer2.source.ExtractorMediaPeriod.readData(ExtractorMediaPeriod.java:370)
    at com.google.android.exoplayer2.source.ExtractorMediaPeriod$SampleStreamImpl.readData(ExtractorMediaPeriod.java:631)
    at com.google.android.exoplayer2.BaseRenderer.readSource(BaseRenderer.java:277)
    at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.feedInputBuffer(MediaCodecRenderer.java:640)
    at com.google.android.exoplayer2.mediacodec.MediaCodecRenderer.render(MediaCodecRenderer.java:530)
    at com.google.android.exoplayer2.ExoPlayerImplInternal.doSomeWork(ExoPlayerImplInternal.java:574)
    at com.google.android.exoplayer2.ExoPlayerImplInternal.handleMessage(ExoPlayerImplInternal.java:350)
    at android.os.Handler.dispatchMessage(Handler.java:98)
    at android.os.Looper.loop(Looper.java:136)
    at android.os.HandlerThread.run(HandlerThread.java:61)
```

Can you give me some ideas? thanks

**Comment by tangmingscau** [comment]
Finally, I found that the error occurred at mediacodec, the system mediacodec got the byteBuffer 37268, but the FLAC music frame data was 49152. So there was a mistake.

**Comment by ojw28** [comment]
Please can you provide a complete bug report (captured with `adb bugreport`) shortly after reproducing the issue. And the content that can be used to reproduce the problem. And information about the device you are reproducing the issue on. All of which are requested in the issue template... Thanks.

**Comment by ojw28** [comment]
A further occurrence of this issue is described (with bug report) in https://github.com/google/ExoPlayer/issues/3648.

**Comment by ojw28** [comment]
Fixed in 2.7.2.
